libpromises/dbm_lmdb.c

```diff
@@ -491,6 +533,220 @@ static void HandleLMDBCorruption(MDB_env *env, const char *msg)
 #endif  /* __MINGW32__ */
 }
 
+/**
+ * A modified clone of HandleLMDBCorruption() for handling full LMDBs. It's not
+ * easy and nice to share much code between the two functions, unfortunately.
+ */
+static void HandleFullLMDB(MDB_env *env)
+{
+    const char *lmdb_file = mdb_env_get_userctx(env);
+    Log(LOG_LEVEL_CRIT, "'%s' DB full!", lmdb_file);
+
+    /* Freeze the DB ASAP. This also makes the call to exit() safe regarding
+     * this particular DB because exit handlers will ignore it. */
+    DBHandle *handle = GetDBHandleFromFilename(lmdb_file);
+    FreezeDB(handle);
+
+#ifdef __MINGW32__
```

Comment:
Yup, we are very inconsistent with this. I'd say right now it doesn't really matter because the moment we want to do non-mingw-based builds, we will have to go through all `__MINGW32__` appearances and check them. But might as well use the proper one in new code. :+1: